backend_kotlin/test-cdc/src/main/kotlin/be/swsb/makeastats/testcdc/HttpBaseConfig.kt

```diff
@@ -0,0 +1,58 @@
+package be.swsb.makeastats.testcdc
+
+import be.swsb.makeastats.kotlinbackend.domain.leaderboard.CreateLeaderBoardCmd
+import be.swsb.makeastats.kotlinbackend.domain.leaderboard.Leaderboard
+import be.swsb.makeastats.kotlinbackend.domain.leaderboard.LeaderboardHashId
+import be.swsb.makeastats.kotlinbackend.domain.leaderboard.LeaderboardRepo
+import be.swsb.makeastats.kotlinbackend.domain.playerstats.PlayerStats
+import be.swsb.makeastats.kotlinbackend.domain.playerstats.PlayerStatsRepo
+import be.swsb.makeastats.kotlinbackend.services.LeaderboardService
+import com.nhaarman.mockito_kotlin.mock
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Primary
+import java.util.*
+import javax.sql.DataSource
+
+@Configuration
+open class HttpBaseConfig {
+
+    @Bean
+    @Primary
+    open fun leaderboardService(): LeaderboardService {
+        return LeaderboardServiceStub(leaderboardRepo(), playerStatsRepo())
+    }
+
+    @Bean
+    @Primary
+    open fun playerStatsRepo(): PlayerStatsRepo {
+        return mock()
+    }
+
+    @Bean
+    @Primary
+    open fun leaderboardRepo(): LeaderboardRepo {
+        return mock()
+    }
+
+    @Bean
+    @Primary
+    open fun datasource(): DataSource {
+        return mock()
+    }
+}
+
+class LeaderboardServiceStub(override val leaderboardRepo: LeaderboardRepo,
```

Comment:
I don't think the repositories need to be mocked. The service could simply be `LeaderboardServiceStub(null, null)` because we  override the default behaviour of the service. Thoughts?

Reply:
In Kotlin you can't _override_ with null values when the type is not a _nullable_ type.

`LeaderboardServiceStub(null,null)` simply wouldn't compile.

Reply:
TIL

Reply:
Kotlin!!! ❤️❤️❤️